.github/workflows/main.yaml

```diff
@@ -171,6 +171,7 @@ jobs:
       - run: make e2e
 
       - uses: actions/upload-artifact@v3
+        if: always()
```

Comment:
It will not upload by default if running successed?

Reply:
no, it will upload logs always,  regardless of success, failure, or cancellation.

Reply:
without this `if: always()` flag, will only upload logs when the last step is successful

Reply:
Oh, so if without `if: always()`, when `make e2e` run failed, this upload-artifact wil be skipped?

Reply:
yes
ref: https://github.com/kubeedge/kubeedge/actions/runs/2501087906